### DISCUSSION TITLE
LDAP Mail Accounts

### DISCUSSION THREAD

**Discussion by humayunha** [the-djmaze/snappymail#1817] [Q&A]
I want to use LDAP mail accounts extension with snappymail.
When extension is not active, I access mail with username. The main reason is to access Nexcloud with username. I want to use LDAP Mail Accounts is to change user and use mail address.
However, I have problems with extension installation. I get request time out error after entering all data. I have attached the settings as an image. Is there a more detailed source and document on this topic? Is anyone else using this extension? If so, can you share the information?
![LDAP](https://github.com/user-attachments/assets/7dc68a85-01fa-4d2d-867d-d9240d6c1915)

**Comment by cm-schl** [comment]
Hello,

>When extension is not active, I access mail with username. The main reason is to access Nexcloud with username. I want to use LDAP Mail Accounts is to change user and use mail address.

Using this plugin you can let the user access additional mail accounts that get retrieved by a ldap query and for those additional accounts you can choose to overwrite the mail address used at login with an other address. Is this that what you're looking for?

Additional documentation can also be found here: https://github.com/the-djmaze/snappymail/tree/master/plugins/ldap-mail-accounts

**Reply by cm-schl** [reply]
I would also like to add that your configuration for the field "LDAP Username" seams wrong - there should be entered a username to login to your LDAP or the field should be empty for anonymous logins.

**Reply by humayunha** [reply]
Hi there,
Thanks for your reply. I solved the problem. I thought it would get the users from BASE DN root, but it was necessary to specify exactly which OU to call the user from. now users in certain OU can connect. but i have problems with users in different OU. when i define the domain admin group with all users, the user connects but the username does not replace the email address. 
![image](https://github.com/user-attachments/assets/4b9d14ad-6ca5-48d3-a092-a58bd8b3a484)

**Reply by cm-schl** [reply]
You set `Username field` to `userPrincipalName` - does this field contains the username (or mail address) used to login to SnappyMail?
Because if I remember correctly this is how it works, the username used to login gets looked up in the field configured in `Username field`.

![grafik](https://github.com/user-attachments/assets/4797a48b-7a4e-4cfd-891e-2bb4581b2413)

**Reply by humayunha** [reply]
username and mail address are different. 
user name: hasimlerm
email: humayun.hasimler@xxx.com

**Reply by cm-schl** [reply]
> You set Username field to userPrincipalName - does this field contains the username (or mail address) used to login to SnappyMail?

So whats inside `userPrincipalName`?

**Reply by humayunha** [reply]
it contains username. 
If I write BASE DN where user is it works perfectly. When Change BASE DN with group after logon do not switch username with email
I connect to the snappymail with user name. I cant connect with email address. This is the next step I have to solv :)
![image](https://github.com/user-attachments/assets/40b3d341-3c88-4078-ad6b-17b3c7e47e14)

**Reply by cm-schl** [reply]
ok thanks.
So in this case you should activate logging (https://github.com/the-djmaze/snappymail/wiki/FAQ#how-do-i-enable-logging) and look up the logs what exactly the plugin is doing...
My doubt is that you're lookup with a BASE DN referencing to a group is not supported or there is something wrong with that. The easiest way would be to get the search strings used by the plugin (written to the logs) and try manually to query your LDAP with the same BASE DN and search string (with some ldap browser for example).

**Reply by humayunha** [reply]
Thanks for your help. Let me try and will give you information.

**Comment by humayunha** [comment]
Hi,
Finally I get logs. When connect to snappy Base DN configured with Domain User Group I get fallowing result. It Connects but does not get email for user.
Used ldap filter to search for mail account(s): (&(objectclass=user)(sAMAccountName=hasimler))
 LDAP MAIL ACCOUNTS PLUGIN[NOTICE]: Could not find user hasimler in LDAP! Overwriting of main mail address not possible.
Base DN: CN=Domain Users,CN=Users,DC=x,DC=y,DC=z

When Connect with specific OU where the user is work porfect.

Used ldap filter to search for mail account(s): (&(objectclass=user)(sAMAccountName=hasimler))
 LDAP MAIL ACCOUNTS PLUGIN[NOTICE]: Attribute xxx not found on object CN=Hasimler  Humayun,OU=x1,OU=User,OU=x2,DC=x,DC=y,DC=z while required
LDAP MAIL ACCOUNTS PLUGIN[WARNING]: Login DATA: login: hasimler email: Humayun.Hasimler@x.com

**Reply by cm-schl** [reply]
Hi,
as your filter is always the same the only things that come in my mind are to check the search query by some type of tool like an LDAP Browser (something like https://www.ldapadministrator.com/softerra-ldap-browser.htm for example). The plugin simply does a search using the filter on the base and gets back some LDAP attributes: https://github.com/cm-schl/snappymail/blob/acaa5bae5514b6851d28978bf4c654015bc83653/plugins/ldap-mail-accounts/LdapMailAccounts.php#L396

My doubt is that the BASE DN points to the wrong position or maybe the LDAP attribute you defined as mail address field does not exist at this base dn... Something like that...
